app/controllers/subscriptions_controller.rb

```diff
@@ -56,4 +63,8 @@ def render_error(message)
       'Por favor, tente novamente mais tarde.'
     redirect_to root_path
   end
+
+  def affiliate_tag
+    request.env['affiliate.tag'] ? request.env['affiliate.tag'] : nil
```

Comment:
This check is completely redundant :P you're basically saying "if `request.env['affiliate.tag']` is `nil` then return `nil`". This does exactly the same as returning `request.env['affiliate.tag']` :)

I believe you meant `request.env['affiliate.tag'].present? ? request.env['affiliate.tag'] : nil`, to check for empty strings as well. This check can be replaced with `request.env['affiliate.tag'].presence`, which [does exactly that](http://api.rubyonrails.org/classes/Object.html#method-i-presence) :)